packages/core/src/types.ts

```diff
@@ -1671,19 +1671,8 @@ export type ActorRefWithDeprecatedState<
   state: State<TContext, TEvent, any, TTypestate, TResolvedTypesMeta>;
 };
 
-export type ActorRefFrom<T> = T extends StateMachine<
-  infer TContext,
-  any,
-  infer TEvent,
-  infer TTypestate,
-  any,
-  any,
-  any
```

Comment:
The issue was that we were not using `infer TResolvedTypesMeta` here and I could have fixed it with a two-line change (infer here + provide the result to StateMachine below). However, I've chosen to refactor this type to use `ReturnTypeOrValue<T> extends infer R` technique so we don't end up in such problems - as it ensures that we infer from direct types and factories in the very same way.